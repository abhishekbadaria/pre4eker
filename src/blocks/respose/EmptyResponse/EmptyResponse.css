.response {
  width: 100%;
  height: calc(80vh);
  background: #F2F2F2;
  border: 1px dashed #000000;
  display: flex;
  justify-content: center;
  align-items: center;
}

.text {
  width: 30rem;
}