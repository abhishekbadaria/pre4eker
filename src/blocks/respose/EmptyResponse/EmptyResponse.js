import React from "react";
import style from "./EmptyResponse.css";

const EmptyResponse = (props) => {
  return (
    <div className={style.response}>
      <div className={style.text}>
        <p>Pre4eker - это сервис проверки и визуализации акций.</p>
        <ol>
          <li>Получи ответ АПИ Mindbox. Например, через Postman или от клиента</li>
          <li>Введи его в форму</li>
          <li>Проверь то, как настроены акции. </li>
        </ol>
        <p>github: <a href="https://github.com/peter-nikitin/mindbox-pre4eker">pre4eker</a></p>
      </div>
    </div>
  );
};

export default EmptyResponse;
