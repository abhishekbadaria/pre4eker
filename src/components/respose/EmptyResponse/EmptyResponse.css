.response {
  display: flex;
  flex-direction: column;
  justify-content: left;
  align-items: left;
  max-width: 800px;
  height: 100vh;
}

.inline {
  display: flex;
  flex-wrap: wrap;
}

.text {
  width: 40%;
  max-width: 100%;
  padding: 0 1rem 1rem;
  min-width: 400px;
}


/* @media (max-width: 1300px) and (min-width: 800px) {
  .inline {
    width: 40%
  }
} */

@media (max-width: 800px) {
  .inline {}
  .text {
    width: auto;
    padding: 1rem 0;
    min-width: auto;
  }
  .response {
    height: auto;
  }
}