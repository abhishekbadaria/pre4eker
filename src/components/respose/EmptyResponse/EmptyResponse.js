import React from "react";
import logo from "src/assets/logo.svg";
import style from "./EmptyResponse.css";

const EmptyResponse = () => (
  <div className={style.response}>
    <div className={style.text}>
      <p>
        <img src={logo} alt="" />
      </p>
      <p>Визуализация акций</p>
    </div>
    <div className={style.inline}>
      <div className={style.text}>
        <div>
          <h3>Расчет и визуализация </h3>
          <p>
            Чтобы рассчитать акции, сервис выполняет запрос{" "}
            <a href="https://developers.mindbox.ru/docs/processing-calculate">
              {" "}
              "Предварительный расчет заказа"
            </a>{" "}
            к API Mindbox.
          </p>
          <p>Для расчета надо: </p>
          <ol>
            <li>
              Заполнить настройки подключения: эндпоинт, секретный ключ и
              операция
            </li>
            <li>Собрать тело запроса</li>
          </ol>
          <p>
            Тело можно собрать через форму-конструктор или вставить готовый JSON
            на вкладке "тело запроса".
          </p>
          <p>Полученный ответ отобразится в виде блоков.</p>
        </div>
      </div>
      <div className={style.text}>
        <h3>Только визуализация</h3>
        <p>
          Если у вас есть расчитанные акции, то вставьте их на вкладке "тело
          ответа".
        </p>
        <p> Заполнять настройки подключения не нужно.</p>
      </div>
    </div>
    <div className={style.text}>
      <h4>Если у вас есть XML</h4>
      <p>
        API Mindbox работает c XML и JSON, но сервис Pre4eker использует только
        JSON.
      </p>
      <p>
        Если у вас есть тело запроса или ответа в формате XML, то его можно
        вставить в соответствующей вкладке. После этого ваш запрос автоматически
        конвертируется в JSON.
      </p>
    </div>
    <div className={style.text}>
      <p>
        github: 
        <a href="https://github.com/peter-nikitin/mindbox-pre4eker">pre4eker</a>
      </p>
    </div>
  </div>
);

export default EmptyResponse;
