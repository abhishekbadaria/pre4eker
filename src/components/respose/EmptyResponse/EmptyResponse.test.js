import React from "react";
import { render, cleanup } from "@testing-library/react";
import EmptyResponse from "./EmptyResponse";

afterEach(cleanup);

test("EmptyResponse has to match snapshot", () => {
  const { asFragment } = render(<EmptyResponse />);
  expect(asFragment(<EmptyResponse />)).toMatchSnapshot();
});
