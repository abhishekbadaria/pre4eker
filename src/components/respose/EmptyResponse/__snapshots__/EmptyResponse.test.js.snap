// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyResponse has to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="response"
  >
    <div
      class="text"
    >
      <p>
        <img
          alt=""
          src="test-file-stub"
        />
      </p>
      <p>
        Визуализация акций
      </p>
    </div>
    <div
      class="inline"
    >
      <div
        class="text"
      >
        <div>
          <h3>
            Расчет и визуализация 
          </h3>
          <p>
            Чтобы рассчитать акции, сервис выполняет запрос 
            <a
              href="https://developers.mindbox.ru/docs/processing-calculate"
            >
               "Предварительный расчет заказа"
            </a>
             к API Mindbox.
          </p>
          <p>
            Для расчета надо: 
          </p>
          <ol>
            <li>
              Заполнить настройки подключения: эндпоинт, секретный ключ и операция
            </li>
            <li>
              Собрать тело запроса
            </li>
          </ol>
          <p>
            Тело можно собрать через форму-конструктор или вставить готовый JSON на вкладке "тело запроса".
          </p>
          <p>
            Полученный ответ отобразится в виде блоков.
          </p>
        </div>
      </div>
      <div
        class="text"
      >
        <h3>
          Только визуализация
        </h3>
        <p>
          Если у вас есть расчитанные акции, то вставьте их на вкладке "тело ответа".
        </p>
        <p>
           Заполнять настройки подключения не нужно.
        </p>
      </div>
    </div>
    <div
      class="text"
    >
      <h4>
        Если у вас есть XML
      </h4>
      <p>
        API Mindbox работает c XML и JSON, но сервис Pre4eker использует только JSON.
      </p>
      <p>
        Если у вас есть тело запроса или ответа в формате XML, то его можно вставить в соответствующей вкладке. После этого ваш запрос автоматически конвертируется в JSON.
      </p>
    </div>
    <div
      class="text"
    >
      <p>
        github: 
        <a
          href="https://github.com/peter-nikitin/mindbox-pre4eker"
        >
          pre4eker
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
